datamodel_code_generator/format.py

```diff
@@ -167,12 +170,49 @@ def __init__(
                 settings_path=self.settings_path, **self.isort_config_kwargs
             )
 
+        self.custom_formatters_kwargs = custom_formatters_kwargs or {}
+        self.custom_formatters = self._check_custom_formatters(custom_formatters)
+
+    def _load_custom_formatter(
+        self, custom_formatter_import: str
+    ) -> CustomCodeFormatter:
+        import_ = import_module(custom_formatter_import)
+
+        if not hasattr(import_, 'CodeFormatter'):
+            raise NameError(
+                f'Custom formatter module `{import_.__name__}` must contains object with name Formatter'
+            )
+
+        formatter_class = import_.__getattribute__('CodeFormatter')
+
+        if not issubclass(formatter_class, CustomCodeFormatter):
+            raise TypeError(
+                f'The custom module {custom_formatter_import} must inherit from `datamodel-code-generator`'
+            )
+
+        return formatter_class(formatter_kwargs=self.custom_formatters_kwargs)
+
+    def _check_custom_formatters(
+        self, custom_formatters: Optional[List[str]]
+    ) -> List[CustomCodeFormatter]:
+        if custom_formatters is None:
+            return []
+
+        return [
+            self._load_custom_formatter(custom_formatter_import)
+            for custom_formatter_import in custom_formatters
+        ]
+
     def format_code(
         self,
         code: str,
     ) -> str:
         code = self.apply_isort(code)
         code = self.apply_black(code)
+
+        for formatter in self.custom_formatters:
+            code = formatter.apply(code)
+
```

Comment:
Should the method apply all formatter?
Someone may want to replace black/isort formatters with others like ruff.

Reply:
> Should the method apply all formatter?

I also agree. I think that in the future we need to add popular custom formatters and freedom of choice.

I think that this method should apply the list of formatters specified by the user like this
```bash
datamodel-codegen ... \
  --formatters \
  "datamodel_code_generator.ruff,user_formatter_module"
```

Reply:
> ... apply the list of formatters specified by the user ...

or default formatter (ruff) or disable format (special flag)

Reply:
OK, What do you think about the option?
```
--default-formatter  [black-isort(default), ruff, none]
```` 
`none` maybe not clearly?
Or
```
--default-formatter  [black-isort(default), ruff]
and 
--disable-default-formatter 
````

Reply:
The second option seems clearer to me.

Then we have 4 args:

`--default-formatter`: choice one (or a few ?) default formatters;
`--disable-default-formatter`: disable default formatters
`--custom-formatters`: user formatters
`--custom-formatters-kwargs`: json file with params for user formatters

Reply:
Looks good.
We have discussed generating non-python code by the CLI.
We should accept another language formatter or turn off the default formatter.
https://github.com/koxudaxi/datamodel-code-generator/discussions/1657

Reply:
> generating non-python code by the CLI

It is good idea! I have a few questions about this.
I will take a look and answer in the discussion.

Reply:
@koxudaxi 

I answer here. I agree with [#1657](https://github.com/koxudaxi/datamodel-code-generator/discussions/1657).
I'm thinking about a more universal approach myself. I have some thoughts about realization details.
I'll try to move in this direction in the next commits.

Reply:
Maybe we can create a discussion where we discuss "road map" for `datamodel-code-generator` version `1.0.0`?

Reply:
I just created it.
https://github.com/koxudaxi/datamodel-code-generator/discussions/1736